packages/integrations/mdx/src/server.ts

```diff
@@ -53,15 +53,14 @@ function throwEnhancedErrorIfMdxComponent(error: Error, Component: any) {
 	// if the exception is from an mdx component
 	// throw an error
 	if (Component[Symbol.for('mdx-component')]) {
-		// if it's an AstroUserError, we don't need to re-throw, keep the original hint
-		if (AstroUserError.is(error)) return;
-		throw new AstroError({
-			message: error.message,
-			title: error.name,
-			hint: `This issue often occurs when your MDX component encounters runtime errors.`,
-			name: error.name,
-			stack: error.stack,
-		});
+		// if it's an existing AstroError, we don't need to re-throw, keep the original hint
+		if (AstroError.is(error)) return;
+		// Mimic the fields of the internal `AstroError` class (not from `astro/errors`) to
+		// provide better title and hint for the error overlay
+		(error as any).title = error.name;
+		(error as any).hint =
+			`This issue often occurs when your MDX component encounters runtime errors.`;
+		throw error;
```

Comment:
I have to tweak this a bit when migrating it into `@astrojs/mdx`. `AstroError` is now actually `AstroUserError` internally (aliased when exporting). But the actual `AstroError` that we usually use internally can't be accessed.

For now, I try to mimic this so that it somewhat works:
Before:
<img width="1281" alt="Screenshot 2024-08-16 at 2 40 22 PM" src="https://github.com/user-attachments/assets/0562c741-b861-4ec7-9aec-567fd1e45f25">

After:
<img width="1266" alt="Screenshot 2024-08-16 at 2 54 02 PM" src="https://github.com/user-attachments/assets/5d360ff4-ca2e-4642-bc01-b2a922b6d552">
(the file name now has line/col, but unfortunately not accurate, like the stack trace. i'm not sure how that happens)

Reply:
Fine with me!